packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -2315,12 +2330,19 @@ export class StripeHelper extends StripeHelperBase {
         }
       }
 
-      const { discount } = sub!;
+      const { discount } = sub;
+
+      // This type inconsistency runs quite deep, but plan does exist on the subscription here
+      // for all current use-cases.
+      const plan = (sub as any).plan as Stripe.Plan;
```

Comment:
This would involve a fairly significant rewrite - I don't know that it's ticketable work at the moment. I spoke to Ben about this last week, but this is going to be part of SubPlat 3.0.